gateway-js/src/supergraphManagers/UplinkFetcher/index.ts

```diff
@@ -80,28 +130,51 @@ export class UplinkFetcher implements SupergraphManager {
     };
   }
 
-  private async updateSupergraphSdl() {
-    const result = await loadSupergraphSdlFromUplinks({
-      graphRef: this.config.graphRef,
-      apiKey: this.config.apiKey,
-      endpoints: this.config.uplinkEndpoints,
-      errorReportingEndpoint: this.errorReportingEndpoint,
-      fetcher: this.config.fetcher,
-      compositionId: this.compositionId ?? null,
-      maxRetries: this.config.maxRetries,
-      roundRobinSeed: this.fetchCount++,
-      earliestFetchTime: this.earliestFetchTime,
-    });
-
-    if (!result) {
-      return null;
-    } else {
+  public get uplinkEndpoints(): string[] {
+    return this._uplinkEndpoints;
+  }
+
+  private async updateSupergraphSdl(): Promise<{supergraphSdl: string, minDelaySeconds?: number} | null> {
+    let supergraphSdl;
+    let minDelaySeconds: number | undefined = this.pollIntervalMs / 1000;
+
+    try {
+      const result = await loadSupergraphSdlFromUplinks({
+        graphRef: this.graphRef,
+        apiKey: this.apiKey,
+        endpoints: this.uplinkEndpoints,
+        errorReportingEndpoint: this.errorReportingEndpoint,
+        fetcher: this.fetcher,
+        compositionId: this.compositionId ?? null,
+        maxRetries: this.maxRetries,
+        roundRobinSeed: this.fetchCount++,
+        earliestFetchTime: this.earliestFetchTime,
+        logger: this.logger
+      });
+
+      if (!result) {
+        return null;
+      }
+
       this.compositionId = result.id;
-      // the healthCheck fn is only assigned if it's enabled in the config
-      await this.healthCheck?.(result.supergraphSdl);
-      const { supergraphSdl, minDelaySeconds } = result;
-      return { supergraphSdl, minDelaySeconds };
+
+      ({supergraphSdl, minDelaySeconds} = result);
+    } catch (e) {
+      if (!this.onFailureToUpdateSupergraphSdl) {
+        throw e;
+      }
+
+      this.logger.debug('Error fetching supergraphSdl from Uplink, calling updateSupergraphSdlFailureCallback');
+      supergraphSdl = await this.onFailureToUpdateSupergraphSdl.call(this, { error: e });
+      // if (!supergraphSdl) {
+      //   throw new UplinkFetcherError('updateSupergraphSdlFailureCallback returned invalid supergraphSdl');
```

Comment:
intentionally left commented?

Reply:
This goes back to what kind of checking and error message we want https://github.com/apollographql/federation/pull/1812#discussion_r862858907

Only checking `!supergraphSdl` doesn't do enough. So I think we're better off not doing any checks here and letting the errors bubble up from `Gateway.updateWithSupergraphSdl`. We _could_ export some of the validation stuff from `Gateway.createSchemaFromSupergraphSdl`, but I'm not sure it's helpful to get the same error at a different part of the call chain.

**TL;DR** This will likely be removed and possibly replaced with more rigorous schema checking.

Reply:
Removed this bit, since checking is done further down the path in https://github.com/apollographql/federation/pull/1812/files#diff-1a556fe2fd6a545a37d601c8185534fec07e931e2599b32d71c767cd149df20eR447